### PR TITLE
Add API procedures to Managing Packages

### PR DESCRIPTION
**Pull request by adamlazik1**
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (planned Satellite 6.14)
* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (Satellite 6.13)
* [x] Foreman 3.4/Katello 4.6 (EL8 only)
* [x] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* [x] Foreman 3.2/Katello 4.4 on EL7 & EL8
* [x] Foreman 3.1/Katello 4.3 on EL7 & EL8 (Satellite 6.11 EL7/8; orcharhino 6.3 on EL7/8)
* We do not accept PRs for Foreman older than 3.1.

### CODE REVIEW

File: guides/common/modules/snip_managing-packages-api.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+Params: {
+    "job_invocation" : {
+                     "inputs" : {
+            "package" : "nano"
+        },
+                        "ssh" : {},
+                 "scheduling" : {},
+        "concurrency_control" : {}, // FIXME: Should this me mentioned in the docs?
```
Comment (adamruzicka) [inline guides/common/modules/snip_managing-packages-api.adoc:25]: Probably, while we're at it

File: guides/common/modules/snip_managing-packages-api.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+. Send a `POST` request with the created body to `/api/job_invocations`.
+
+.An Example API Request
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+Params: {
+    "job_invocation" : {
+                     "inputs" : {
+            "package" : "nano"
```
Comment (adamruzicka) [inline guides/common/modules/snip_managing-packages-api.adoc:21]: It should probably say that multiple packages can be delimited by whitespace

**Review by adamruzicka — COMMENTED** [review]
I'd be a little bit more specific about types and distinguish between `field`, `array`, `object` and so on

File: guides/common/modules/snip_managing-packages-api.adoc [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        },
+                        "ssh" : {},
+                 "scheduling" : {},
+        "concurrency_control" : {
+           "concurrency_level": "100"
```
Comment (adamruzicka) [inline guides/common/modules/snip_managing-packages-api.adoc:29]: This should be an integer, not a string

File: guides/common/modules/snip_managing-packages-api.adoc [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+* Static query evaluates immediately when accpeting the API request.
+* Dynamic query evaluates when the job is run.
+This evaluation does not occur at the time of acceping the API request if the job is scheduled to run at a later time.
+
+. Send a `POST` request with the created body to `/api/job_invocations`.
+
+.An Example API Request
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+Params: {
```
Comment (ekohl) [inline guides/common/modules/snip_managing-packages-api.adoc:21]: I don't think `Params:` is part of the JSON. Also, is there JSON syntax highlighting?

File: guides/common/modules/snip_managing-packages-api.adoc [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                    "feature" : "katello_package_{PackageAction}",
+                    "ansible" : {}, // FIXME: Should this be mentioned in the docs?
+             "targeting_type" : "static_query"
+    }
+}
+----
+
+[NOTE]
+====
+You do not need to include empnty arrays in the body of your API request.
```
Comment (ekohl) [inline guides/common/modules/snip_managing-packages-api.adoc:41]:
```suggestion
You do not need to include empty arrays in the body of your API request.
```

Though I only see what JSON calls objects (`{}`), though I admit I had to look that up.

File: guides/common/modules/snip_managing-packages-api.adoc (hunk 4 above)
Comment (adamlazik1) [reply guides/common/modules/snip_managing-packages-api.adoc]: Removed.

File: guides/common/modules/snip_managing-packages-api.adoc (hunk 5 above)
Comment (adamlazik1) [reply guides/common/modules/snip_managing-packages-api.adoc]: Removed the note as it is no longer necessary.

File: guides/common/modules/snip_managing-packages-api.adoc [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                 "scheduling" : {
+                    "start_at": "2023-09-21T19:00:00+00:00",
+                "start_before": "2023-09-23T00:00:00+00:00"
+                 },
+        "concurrency_control" : {
+           "concurrency_level": 100
+        },
+             "targeting_type" : "dynamic_query"
+    }
+}
```
Comment (adamruzicka) [inline guides/common/modules/snip_managing-packages-api.adoc:66]: To be honest I'm not a huge fan of this colon-aligned format, could we do two spaces per level?

```suggestion
{
   "job_invocation" : {
      "concurrency_control" : {
         "concurrency_level" : 100
      },
      "feature" : "katello_package_{PackageAction}",
      "inputs" : {
         "package" : "nano vim"
      },
      "scheduling" : {
         "start_at" : "2023-09-21T19:00:00+00:00",
         "start_before" : "2023-09-23T00:00:00+00:00"
      },
      "search_query" : "*",
      "ssh" : {
         "effective_user" : "My_Username",
         "effective_user_password" : "My_Password"
      },
      "targeting_type" : "dynamic_query"
   }
}
```

**Review by adamruzicka — COMMENTED** [review]
Looks good content-wise

File: guides/common/modules/snip_managing-packages-api.adoc (hunk 6 above)
Comment (adamlazik1) [reply guides/common/modules/snip_managing-packages-api.adoc:66]: For sure. To be honest, I just kept the format hammer threw at me :sweat_smile:

**Comment by adamlazik1** [comment]
Could some please writer review this?

File: guides/common/modules/snip_managing-packages-api.adoc [hunk 7]
```diff
@@ -0,0 +1,67 @@
+.API Procedure
+Create a body of the API request in the JSON format by following the instructions below.
+
```
Comment (maximiliankolb) [inline guides/common/modules/snip_managing-packages-api.adoc:3]:
```suggestion
Create a body of the API request in the JSON format by following the instructions below.

.API Procedure
```

File: guides/common/modules/snip_managing-packages-api.adoc [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+-H "content-type: application/json" \
+-X POST \
+-d @_Path_To_My_API_Request_Body_ \
+-u _My_Username_:__My_Password__ | python3 -m json.tool
+----
+
+.Verification
+* In the {ProjectWebUI}, navigate to *Monitor* > *Jobs* and see the report of the scheduled or completed remote execution job to {PackageAction} the packages on the selected hosts.
+
+.An Example API Request Body
```
Comment (maximiliankolb) [inline guides/common/modules/snip_managing-packages-api.adoc:43]:
```suggestion
.Example API Request Body
```

File: guides/common/modules/snip_managing-packages-api.adoc [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+[options="nowrap", subs="+quotes,verbatim,attributes"]
```
Comment (maximiliankolb) [inline guides/common/modules/snip_managing-packages-api.adoc:44]:
```suggestion
[options="nowrap", json, subs="+quotes,verbatim,attributes"]
```

Unsure, but something like this _should_ work. Feel free to drop this suggestion.

**Review by maximiliankolb — COMMENTED** [review]
Some minor comments on style. Rest LGTM

File: guides/common/modules/snip_managing-packages-api.adoc (hunk 9 above)
Comment (adamlazik1) [reply guides/common/modules/snip_managing-packages-api.adoc:44]: I'll give it a try.

Comment (maximiliankolb) [reply guides/common/modules/snip_managing-packages-api.adoc]: definitely *not* blocking your PR though!

File: guides/common/modules/snip_managing-packages-api.adoc [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+-X POST \
+-d @_Path_To_My_API_Request_Body_ \
+-u _My_Username_:__My_Password__ | python3 -m json.tool
+----
+
+.Verification
+* In the {ProjectWebUI}, navigate to *Monitor* > *Jobs* and see the report of the scheduled or completed remote execution job to {PackageAction} the packages on the selected hosts.
+
+.Example API Request Body
+[options="nowrap", json, subs="+quotes,verbatim,attributes"]
```
Comment (maximiliankolb) [inline guides/common/modules/snip_managing-packages-api.adoc:44]: Merge with or without `json`.

**Review by maximiliankolb — APPROVED** [review]
LGTM!

File: guides/common/modules/snip_managing-packages-api.adoc (hunk 10 above)
Comment (adamlazik1) [reply guides/common/modules/snip_managing-packages-api.adoc:44]: it seems to add a little json mark on the code block so I will keep it.

**Comment by adamlazik1** [comment]
Merged and cherry-picked:
   8d6bbb2a6..76abe50a6  3.3 -> 3.3
   855c70a1d..812f90bc2  3.4 -> 3.4
   0f50baaa3..629c3f1d3  3.5 -> 3.5
   bcae9b56d..d45425586  3.6 -> 3.6
   5355e6836..8a6a88456  3.7 -> 3.7
   cad40f034..457a94847  3.8 -> 3.8

Merge conflicts on 3.2 and 3.1, but those will become unsupported on satellite with 6.14 release so I will not cherry-pick there.

**Pull request merged**